### PR TITLE
Bug 1984365: Dashboard Prometheus/Overview can't filter instance by job

### PR DESCRIPTION
**Pull request by PhilipGough**
EDIT: reduced scope of this PR to only include issues related to the bug.

* [x] I added CHANGELOG entry for this change.

### CODE REVIEW

File: pkg/tasks/prometheus.go [hunk 1]
```diff
@@ -235,6 +235,11 @@ func (t *PrometheusTask) Run(ctx context.Context) error {
 		return errors.Wrap(err, "reconciling Prometheus rules PrometheusRule failed")
 	}
 
+	_, err = t.factory.PrometheusK8sThanosSidecarPrometheusRule()
+	if err != nil {
+		return errors.Wrap(err, "initializing Prometheus rules for Thanos Sidecar failed")
+	}
+
```
Comment (prashbnair) [inline pkg/tasks/prometheus.go:242]: Should this rule not be created when CMO is deployed?

File: pkg/tasks/prometheus.go [hunk 2]
```diff
@@ -235,6 +235,16 @@ func (t *PrometheusTask) Run(ctx context.Context) error {
 		return errors.Wrap(err, "reconciling Prometheus rules PrometheusRule failed")
 	}
 
+	tsRule, err := t.factory.PrometheusK8sThanosSidecarPrometheusRule()
+	if err != nil {
+		return errors.Wrap(err, "initializing Prometheus rules for Thanos Sidecar failed")
```
Comment (simonpasquier) [inline pkg/tasks/prometheus.go:240]:
```suggestion
		return errors.Wrap(err, "initializing Thanos Sidecar rules failed")
```

File: pkg/tasks/prometheus.go [hunk 3]
```diff
@@ -235,6 +235,16 @@ func (t *PrometheusTask) Run(ctx context.Context) error {
...
+	}
+
+	err = t.client.CreateOrUpdatePrometheusRule(ctx, tsRule)
+	if err != nil {
+		return errors.Wrap(err, "reconciling Prometheus rules PrometheusRule failed")
```
Comment (simonpasquier) [inline pkg/tasks/prometheus.go:245]:
```suggestion
		return errors.Wrap(err, "reconciling Thanos Sidecar rules PrometheusRule failed")
```

**Review by simonpasquier — COMMENTED** [review]
could you scope down the update to only the relevant pieces? E.g. this updates `manifests/0000_50_cluster-monitoring-operator_00_0*-custom-resource-definition.yaml` files to v0.50.0 while the prometheus operator version is still v0.49.0.

**Review by paulfantom — APPROVED** [review]
/lgtm

**Pull request merged**